core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -403,11 +407,24 @@ private[log] class Cleaner(val id: Int,
 
     try {
       // clean segments into the new destination segment
-      for (old <- segments) {
+      val iter = segments.iterator
+      var current: Option[LogSegment] = Some(iter.next())
+      while (current.isDefined) {
+        val old = current.get
```

Comment:
Having old, current, next might be a bit confusing. Perhaps current -> currentOpt and old -> current ?